src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php

```diff
@@ -533,6 +534,20 @@ public function testServiceLocator()
                 'nil' => $nil = new ServiceClosureArgument(new Reference('nil')),
             ))
         ;
+
+        $container->register('method_call_class', MethodCallClass::class);
+
+        // 1 callable method
+        $container->register('service_locator_with_inline_reference_1', ServiceLocator::class)
+            ->addArgument(array('method_call_class' => new ServiceClosureArgument(new Reference('method_call_class'))))
+            ->addMethodCall('callableMethod', array('a', new Reference('method_call_class')));
+
+        // 2 callable methods
+        $container->register('service_locator_with_inline_reference_2', ServiceLocator::class)
+            ->addArgument(array('method_call_class' => new ServiceClosureArgument(new Reference('method_call_class'))))
+            ->addMethodCall('callableMethod', array('a', new Reference('method_call_class')))
+            ->addMethodCall('callableMethod', array('b', new Reference('method_call_class')));
```

Comment:
Calling methods on a service locator looks weird to me. It breaks the laziness of the locator (as instantiating the ServiceLocator involves getting some instances in it to call the method). The method call should be added in the definition of the referenced service instead.
I don't see any use case for such calling on the ServiceLocator instance (and so not sure we should allow such feature and spend time fixing edge cases in it).

/cc @nicolas-grekas